strapi/strapi
# Fix unable to delete an image from the media-library

Signed-off-by: harimkims <harimkims@gmail.com>



### What does it do?

- Avoid direct access to created by attribute in file, Use [lodash](https://lodash.com/) get instead.

### Why is it needed?

Because some uploaded files doesn't have `createdBy` attribute - for example when you upload file as a public Role - It throws an error when it does not have author.

### How to test it?

1. Give upload permission to the `Public` role.
2. `curl https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png -o lenna.png`
3. `curl -X 'POST' 'http://localhost:1337/api/upload' -F 'files=@lenna.png'`
4. Go to strapi admin-> media-library
5. Select lenna.png
6. Click on button "Delete"
7. Click on "Confirm" on the popup

### Related issue(s)/PR(s)

Closes #11939
